From: Heiko Stuebner <heiko@sntech.de>
Subject: [PATCH] Disable symbol versioning

libnl3 after 3.2.21 enabled symbol versioning [0] without increasing the
soname of the library, producing issues with our symbol lists.

Until we hit a regular soname bump, it might be easiest to keep symbol
versioning disabled.


[0] https://github.com/thom311/libnl/commit/0eb665c693ab2673b722ffd75df744f75fb7e2fe

Signed-off-by: Heiko Stuebner <heiko@sntech.de>

---
 Makefile.am | 21 +++++++--------------
 1 file changed, 7 insertions(+), 14 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 7747d16..d39fcde 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -336,8 +336,7 @@ EXTRA_lib_libnl_3_la_DEPENDENCIES = \
 lib_libnl_3_la_CPPFLAGS = \
 	$(lib_cppflags)
 lib_libnl_3_la_LDFLAGS = \
-	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
-	-Wl,--version-script=$(srcdir)/libnl-3.sym
+	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
 lib_LTLIBRARIES += lib/libnl-route-3.la
 
@@ -422,8 +421,7 @@ EXTRA_lib_libnl_route_3_la_DEPENDENCIES = \
 lib_libnl_route_3_la_CPPFLAGS = \
 	$(lib_cppflags)
 lib_libnl_route_3_la_LDFLAGS = \
-	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
-	-Wl,--version-script=$(srcdir)/libnl-route-3.sym
+	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 lib_libnl_route_3_la_LIBADD = \
 	lib/libnl-3.la
 
@@ -442,8 +440,7 @@ EXTRA_lib_libnl_idiag_3_la_DEPENDENCIES = \
 lib_libnl_idiag_3_la_CPPFLAGS = \
 	$(lib_cppflags)
 lib_libnl_idiag_3_la_LDFLAGS = \
-	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
-	-Wl,--version-script=$(srcdir)/libnl-idiag-3.sym
+	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 lib_libnl_idiag_3_la_LIBADD = \
 	lib/libnl-3.la
 
@@ -459,8 +456,7 @@ EXTRA_lib_libnl_genl_3_la_DEPENDENCIES = \
 lib_libnl_genl_3_la_CPPFLAGS = \
 	$(lib_cppflags)
 lib_libnl_genl_3_la_LDFLAGS = \
-	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
-	-Wl,--version-script=$(srcdir)/libnl-genl-3.sym
+	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 lib_libnl_genl_3_la_LIBADD = \
 	lib/libnl-3.la
 
@@ -486,8 +482,7 @@ lib_libnl_nf_3_la_CPPFLAGS = \
 EXTRA_lib_libnl_nf_3_la_DEPENDENCIES = \
 	libnl-nf-3.sym
 lib_libnl_nf_3_la_LDFLAGS = \
-	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
-	-Wl,--version-script=$(srcdir)/libnl-nf-3.sym
+	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 lib_libnl_nf_3_la_LIBADD  = \
 	lib/libnl-3.la \
 	lib/libnl-route-3.la
@@ -504,8 +499,7 @@ lib_libnl_xfrm_3_la_SOURCES = \
 lib_libnl_xfrm_3_la_CPPFLAGS = \
 	$(lib_cppflags)
 lib_libnl_xfrm_3_la_LDFLAGS = \
-	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
-	-Wl,--version-script=$(srcdir)/libnl-xfrm-3.sym
+	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 EXTRA_lib_libnl_xfrm_3_la_DEPENDENCIES = \
 	libnl-xfrm-3.sym
 lib_libnl_xfrm_3_la_LIBADD = \
@@ -594,8 +588,7 @@ src_lib_libnl_cli_3_la_CPPFLAGS = \
 	-I$(srcdir)/include \
 	-I$(builddir)/include
 src_lib_libnl_cli_3_la_LDFLAGS = \
-	$(src_lib_ldflags) \
-	-Wl,--version-script=$(srcdir)/libnl-cli-3.sym
+	$(src_lib_ldflags)
 src_lib_libnl_cli_3_la_LIBADD = \
 	lib/libnl-3.la \
 	lib/libnl-route-3.la \
-- 
tg: (38d147c..) debian/no-symvers (depends on: upstream-dist)
